test/unit/test_lexicon.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not
+# use this file except in compliance with the License. A copy of the License
+# is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+import os
+from tempfile import TemporaryDirectory
+
+import mxnet as mx
+import numpy as np
+
+import sockeye.constants as C
+import sockeye.lexicon
+
+
+def test_topk_lexicon():
+
+    lexicon = ["a\ta\t-0.6931471805599453",
+               "a\tb\t-1.2039728043259361",
+               "a\tc\t-1.6094379124341003",
+               "b\tb\t0.0"]
+    vocab_list = ["a", "b", "c"]
+    vocab = dict((y, x) for (x, y) in enumerate(C.VOCAB_SYMBOLS + vocab_list))
+    k = 2
+    lex = sockeye.lexicon.TopKLexicon(vocab, vocab)
+
+    # Create from known lexicon
+    with TemporaryDirectory(prefix="test_topk_lexicon.") as work_dir:
+        input_lex_path = os.path.join(work_dir, "input.lex")
+        with open(input_lex_path, "w") as out:
+            for line in lexicon:
+                print(line, file=out)
```

Comment:
why print here? would it make sense to assert on comparing `line` with an expected line?

Reply:
Looked back and saw this wasn't documented well enough.  We write a fast_align format lex table to the temp dir, then call sockeye.lexicon to build the top-k lexicon from the fast_align lex table.  I added some comments.

Reply:
You are right sorry. I am still getting used to print writing to file descriptors :)